src/app/components/top-nav.tsx

```diff
@@ -0,0 +1,72 @@
+"use client";
+import { Button } from "@/components/ui/button";
+import { Separator } from "@/components/ui/separator";
+import {
+  Sheet,
+  SheetContent,
+  SheetHeader,
+  SheetTrigger,
+} from "@/components/ui/sheet";
+import { cn } from "@/lib/utils";
+import { Bars3Icon, HomeIcon } from "@heroicons/react/20/solid";
+import Link, { LinkProps } from "next/link";
+import { usePathname } from "next/navigation";
+import { useState } from "react";
+
+type MobileLinkProps = LinkProps & {
+  children: React.ReactNode;
+  className?: string;
+  onOpenChange: (open: boolean) => void;
+};
+function MobileLink({ onOpenChange, className, ...props }: MobileLinkProps) {
+  const pathname = usePathname();
+  const isActive = pathname === props.href;
+  return (
+    <Link
+      aria-current={isActive ? "page" : undefined}
+      className={cn(
+        "flex flex-row items-center w-full rounded px-4 py-2 aria-[current='page']:bg-gray-100",
```

Comment:
To continue from above, to further add to mobile user-friendliness, I always like it on mobile when I press a button and it changes color, telling me I've actually pressed it with my fat thumb. I had a go implementing it but it seems to be permanently overrided by the `aria-[current='page']:bg-gray-100` value.
```suggestion
        "flex flex-row items-center w-full rounded px-4 py-2 aria-[current='page']:bg-gray-100 active:bg-gray-200",
```
^ bg-gray-200 as an example, we can choose anything, of course.

Reply:
Ah yeah, this is an unfortunate side effect of the way that Tailwind generates its CSS file. It places `aria-` rules _after_ the `active` rules and so the `aria` always wins out. We can definitely resolve this in some way, either by adjusting when the rule is stored or by setting it to `!important` (which I would generally avoid but could be helpful). 

I do have a couple of questions tho: 

1. Is it useful to have the current page link flash a specific color to let you know it's been hit, even tho hitting it effectively doesn't do anything ('cause it's a link to the page you're already on)
2. Do you think that button flashing a different color is valuable if the navigation then immediately closes? 

I have no strong opinion on these either way but I'm interested in the discussion.